tests/integration/targets/azure_rm_eventhub/tasks/main.yml

```
@@ -0,0 +1,142 @@
+- name: Create random event hub and namespace
+  set_fact:
+      namespace_name: "test{{ resource_group | hash('md5') | truncate(16, True, '') + (65535 | random | string) }}"
+      name: "test{{ resource_group | hash('md5') | truncate(16, True, '') + (65535 | random | string) }}"
+
+- name: Create Event Hub Namespace (check mode)
+  azure_rm_eventhub:
+    location: eastus2
+    namespace_name: "{{ namespace_name }}"
+    resource_group: "{{ resource_group }}"
+    sku: "Basic"
+  check_mode: yes
+  register: results
+
+- assert:
+    that: results.changed
+
+- name: Create Event Hub  (check mode)
+  azure_rm_eventhub:
+    location: eastus2
+    namespace_name: "{{ namespace_name }}"
+    name: "{{ name }}"
+    resource_group: "{{ resource_group }}"
+    sku: "Basic"
+  check_mode: yes
+  register: results
+
+- assert:
+    that: results.changed
+
+- name: Create Event Hub Namespace
+  azure_rm_eventhub:
+    location: eastus2
+    namespace_name: "{{ namespace_name }}"
+    resource_group: "{{ resource_group }}"
+    sku: "Basic"
+  register: results
+
+- assert:
+    that: results.changed
+
+- name: Create Event Hub
+  azure_rm_eventhub:
+    location: eastus2
+    namespace_name: "{{ namespace_name }}"
+    name: "{{ name }}"
+    resource_group: "{{ resource_group }}"
+    sku: "Basic"
+  register: results
+
+- assert:
+    that: results.changed
+
+- name: Update Namespace
+  azure_rm_eventhub:
+    location: eastus2
+    namespace_name: "{{ namespace_name }}"
+    resource_group: "{{ resource_group }}"
+    sku: "Standard"
+    tags:
+      test: modified
+  register: results
+
+- assert:
+    that: 
+      - results.changed
+      - results.state.tags.test == 'modified'
+
+- name: Update Event Hub
+  azure_rm_eventhub:
+    location: eastus2
+    namespace_name: "{{ namespace_name }}"
+    name: "{{ name }}"
+    resource_group: "{{ resource_group }}"
+    message_retention_in_days: 2
+    tags:
+      test: modified
+  register: results
+
+- assert:
+    that: 
+      - not results.changed
+
+- name: Retrieve Namespace
+  azure_rm_eventhub_info:
+    namespace_name: "{{ namespace_name }}"
+    resource_group: "{{ resource_group }}"
+  register: results
+
+- name: Test idempotent
+  azure_rm_eventhub:
+    location: eastus2
+    namespace_name: "{{ namespace_name }}"
+    resource_group: "{{ resource_group }}"
+  register: results
+
+- assert:
+    that:
+      - results.changed
```

Since this is idempotent, it should return results.changed is False

```suggestion
      - not results.changed
```